Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -125,60 +129,80 @@ namespace GradientSignal
                 // imageBoundsY)
                 AZ::Vector3 uvw(static_cast<float>(x), static_cast<float>((imageResolutionY - 1) - y), 0.0f);
 
-                GradientSampleParams sampleParams;
-                sampleParams.m_position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
+                AZ::Vector3 position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
 
                 bool inBounds = true;
                 LmbrCentral::ShapeComponentRequestsBus::EventResult(
-                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, sampleParams.m_position);
+                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, position);
 
-                float sample = inBounds ? m_configuration.m_gradientSampler.GetValue(sampleParams) : 0.0f;
-
-                // Write out the sample value for the pixel based on output format
-                int index = ((y * imageResolutionX) + x) * channels;
-                switch (m_configuration.m_outputFormat)
-                {
-                case OutputFormat::R8:
+                if (!inBounds)
                 {
-                    AZ::u8 value = static_cast<AZ::u8>(sample * std::numeric_limits<AZ::u8>::max());
-                    pixels[index] = value; // R
-
-                    if (channels == 4)
-                    {
-                        pixels[index + 1] = value; // G
-                        pixels[index + 2] = value; // B
-                        pixels[index + 3] = std::numeric_limits<AZ::u8>::max(); // A
-                    }
-                    break;
+                    continue;
                 }
-                case OutputFormat::R16:
+
+                // Keep track of this input position + the x,y indices
+                inputPositions.push_back(position);
+                indices.push_back(AZStd::make_pair(x, y));
+            }
+        }
+
+        // Retrieve all the gradient values for the input positions
+        const size_t numPositions = inputPositions.size();
+        AZStd::vector<float> outputValues(numPositions);
+        m_configuration.m_gradientSampler.GetValues(inputPositions, outputValues);
+
+        // Write out all the gradient values to our output image
+        for (int i = 0; !m_shouldCancel && (i < numPositions); ++i)
+        {
+            const float& sample = outputValues[i];
+            const auto& indicesPair = indices[i];
+            const auto& x = indicesPair.first;
+            const auto& y = indicesPair.second;
```

Comment:
nit: you could use a structured binding to make this a little prettier - `const auto& [x, y] = indices[i];`

Reply:
Ah thanks I knew there was a shorthand for this but forgot, will update!